proposals/support_different_argo_wf_versions_testplan/different_aw_test_plan.md

```diff
@@ -0,0 +1,47 @@
+| **Expected Results**  | <ul><li>Full compatibility with current version</li><li> All pipeline features operational</li><li> No breaking changes or issues</li><li> Performance within acceptable range </li>                                                                                             |
+
+### 2. Previous Version (N-1) Compatibility
+
+| Test Case ID          | TC-CM-002                                                                                                                                                                                                                                   |
+|-----------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **Test Case Summary** | Validate compatibility with previous supported Argo version                                                                                                                                                                                 |
+| **Test Steps**        | <ol><li> Install previous supported Argo version (e.g., 3.5.y) </li><li>Configure KFP with this Argo version</li><li>Execute pipeline run</li><li>Document compatibility differences</li><li>Verify core functionality maintained</li></ol> |
+| **Expected Results**  | <ul><li>Core functionality works with N-1 version</li><li> Any limitations clearly documented</li><li> No critical failures or data loss</li><li> Upgrade path available  </li>                                                             |
+
+### 2.1 Z-Stream Version Compatibility
+
+| Test Case ID          | TC-CM-002a                                                                                                                                                                                                                                                                                                                                                                  |
+|-----------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **Test Case Summary** | Validate compatibility with z-stream (patch) versions of Argo                                                                                                                                                                                                                                                                                                               |
```

Comment:
I think at some point we have to trust that Argo Workflow is not breaking api compatibility in patch releases, and I'm skeptical that we should add this, if it's in non-PR CI that runs nightly or something, I guess that's fine